### PR TITLE
Correctly process `error.Redirect` exceptions

### PR DESCRIPTION
**Pull request by sveneberth**

### CODE REVIEW

**Comment by sveneberth** [comment]
Hey there,
IMO we should redirect to https as 301. Any objections?

https://github.com/viur-framework/server/blob/e53c744834103a467a5f51b6b29db7aa531a6096/__init__.py#L446
would become
```python
self.redirect( e.url.encode("UTF-8"), status=301 ) 
```

File: errors.py [hunk 1]
```diff
@@ -38,7 +38,7 @@ class Redirect( HTTPException ):
 		Causes an 303 - See Other (or 302 - Found if requested / 301 - Moved Permanently) redirect
 	"""
 	def __init__( self, url, descr="Redirect", status=303 ):
-		super( Redirect, self ).__init__(  status=303, name = "Redirect", descr=descr )
+		super( Redirect, self ).__init__(  status=status, name="Redirect", descr=descr )
```
Comment (phorward) [inline errors.py]: you might check for approriate status codes here, so nobody is a ble to send status 200 or something else via errors.Redirect()

```python
if status not in [301, 302, 303]: ..
```
I'm not sure if its better to let status default to 301 in this case or to raise an exception out of this exception...

**Review by phorward — CHANGES_REQUESTED** [review]
please add the status code checkup and run a code reformat on your added code.

**Review by phorward — APPROVED** [review]

**Comment by tsteinruecken** [comment]
Too much unrelated changes in this PR. 2.X is EOL, keep PRs as small as possible. Also, there is only a fixed set of valid redirect status codes (301, 302, 303, 307 & 308) - others have a different meaning and must be treated differntly

**Comment by phorward** [comment]
> Also, there is only a fixed set of valid redirect status codes (301, 302, 303, 307 & 308) - others have a different meaning and must be treated differntly

Can't @sveneberth do another PR on this with lesser changes (code formatting?). I'll reopen this, most PRs here had been made for a reason, theres work within, and it's fair to discuss about it, what the intention behind a PR was and how the PR can be integrated.

**Comment by sveneberth** [comment]
> Too much unrelated changes in this PR. 2.X is EOL, keep PRs as small as possible.

Hmm... I wouldn't like to start with each PR a new discussion about how to treat features/fixes/... for 2.x. - IMO the 2.x. should receive further improvements and bugfixes except breaking changes. Therefore we have to many project running on this major version.


> Also, there is only a fixed set of valid redirect status codes (301, 302, 303, 307 & 308) - others have a different meaning and must be treated differntly

As I understood this is incorrect. All 3xx are valid for Redirects and do in general the same. Of course are only the above mentioned codes common. But there are no restrictions to use other.

Here I refer to the [RFC 7231](https://tools.ietf.org/html/rfc7231#section-6):
> "**HTTP status codes are extensible**.  HTTP clients are not required to
   understand the meaning of all registered status codes, though such
   understanding is obviously desirable.  However, a client MUST
   **understand the class of any status code**, as indicated by the first
   digit, and treat an unrecognized status code as being equivalent to
   the x00 status code of that class"

**Comment by tsteinruecken** [comment]
For reference: This has been discussed at the last ViUR meeting; i'll close this in favor of a new, clean PR. Unknown 3XX-Status-Codes will not act as a redirect as expected, as the fallback HTTP 300 is Multiple Choices.

**Comment by sveneberth** [comment]
Replaced by #198

**Pull request closed without merging**